### PR TITLE
Contributing docs: Add caveat about direct subdirectories and exec tests

### PR DESCRIPTION
**Pull request by maurobringolf**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | no
| Patch: Bug Fix?          | no
| Major: Breaking Change?  | no
| Minor: New Feature?      | no
| Tests Added + Pass?      | No
| Documentation PR         | Yes
| Any Dependency Changes?  | no
| License                  | MIT

In https://github.com/babel/babel/pull/7475 I discovered that we cannot have `input.js`-`output.js` tests in first level child-directories of `fixtures`. For example `fixtures/some/input.js` is run as exec-test and no `output.js` is generated for it. This PR adds a note to the contributing docs for that.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -175,8 +175,8 @@ For example, in [`@babel/plugin-transform-exponentiation-operator/test`](https:/
...
-   - If you need to expect an error, you can ignore creating the `output.js` file and pass a new `throws` key to the `options.json` that contains the error string that is created.
+   - The first is a simple test of the input and output produced by running Babel on some code. We do this by creating an [`input.js`](https://github.com/babel/babel/blob/master/packages/babel-plugin-transform-exponentiation-operator/test/fixtures/exponentian-operator/binary/input.js) file and an [`output.js`](https://github.com/babel/babel/blob/master/packages/babel-plugin-transform-exponentiation-operator/test/fixtures/exponentian-operator/binary/output.js) file. This kind of test _only works in sub-subdirectories of `/fixtures`, i.e. `/fixtures/exponentian-operator/binary/input.js` and **not** `/fixtures/exponentian-operator/input.js`.
```
Comment (nicolo-ribaudo) [inline CONTRIBUTING.md]: You forgot _ after only.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -175,8 +175,8 @@ For example, in [`@babel/plugin-transform-exponentiation-operator/test`](https:/
 
 - In each subfolder, you can organize your directory structure by categories of tests. (Example: these folders can be named after the feature you are testing or can reference the issue number they fix)
 - Generally, there are two kinds of tests for plugins
-   - The first is a simple test of the input and output produced by running Babel on some code. We do this by creating an [`input.js`](https://github.com/babel/babel/blob/master/packages/babel-plugin-transform-exponentiation-operator/test/fixtures/exponentian-operator/binary/input.js) file and an [`output.js`](https://github.com/babel/babel/blob/master/packages/babel-plugin-transform-exponentiation-operator/test/fixtures/exponentian-operator/binary/output.js) file.
-   - If you need to expect an error, you can ignore creating the `output.js` file and pass a new `throws` key to the `options.json` that contains the error string that is created.
```
Comment (nicolo-ribaudo) [inline CONTRIBUTING.md]: Why did you remove this indentation?

Comment (maurobringolf) [reply CONTRIBUTING.md]: Not sure what happened here, I didnt mean to change any indentation

**Comment by nicolo-ribaudo** [comment]
Thank you!
